cmd/wallet-web/src/components/Signout/SignoutComponent.vue

```diff
@@ -29,17 +26,21 @@ const signoutUser = () => store.dispatch('logout');
 async function signout() {
   const actions = [signoutUser()];
   if (isCHAPI.value) {
+    const polyfill = await import('credential-handler-polyfill');
+    const webCredentialHandler = await import('web-credential-handler');
+    const agentOpts = computed(() => store.getters['getAgentOpts']);
+
+    const chapi = ref(
+      new CHAPIHandler(polyfill, webCredentialHandler, agentOpts.value.credentialMediatorURL)
+    );
+
     actions.push(chapi.value.uninstall());
```

Comment:
can we do it in better way?
triggering an vuex action and that action finds CHAPI if it's there and uninstall it.

I would suggest all CHAPI handling (install, register, uninstall) isolated and called only by concerned actions.

Reply:
This is the logic we used up until now, I just added a conditional statement here to support the flag feature. I totally agree with your suggestion to clean this up, but maybe we can address it after we overview wallet data/vuex model and architecture?